### ISSUE TITLE
How do you use `BeApproximately` with a value that changes in each iteration of a loop?

### ISSUE THREAD

**Issue by Smurf-IV** [fluentassertions/fluentassertions#1384]
### Description

- I'm iterating over 2 lists
- I want to compare the 2 values at each point in the list
- I want the value to be within 0.1% of each other

### Complete minimal example reproducing the issue

```csharp
var sheetRows = new List<float> { -24.0, 0.1, 10};
var loads= new List<float> { -24.0, 0.1, 10.05};
foreach ((var xl, var db) in sheetRows.Zip(loads, (n, w) => (n, w)))
{
   xl.Should().BeApproximately(db, db*0.001f);
}
```

### Expected behavior:

To pass

### Actual behavior:
```
Expected xl to approximate -24F +/- -0.024F, but -24F differed by 0F.
   at FluentAssertions.Execution.LateBoundTestFramework.Throw(String message)
   at FluentAssertions.Execution.TestFrameworkProvider.Throw(String message)
   at FluentAssertions.Execution.DefaultAssertionStrategy.HandleFailure(String message)
   at FluentAssertions.Execution.AssertionScope.FailWith(Func`1 failReasonFunc)
   at FluentAssertions.Execution.AssertionScope.FailWith(Func`1 failReasonFunc)
   at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args)
   at FluentAssertions.NumericAssertionsExtensions.FailIfDifferenceOutsidePrecision[T](Boolean differenceWithinPrecision, NumericAssertions`1 parent, T expectedValue, T precision, T actualDifference, String because, Object[] becauseArgs)
   at FluentAssertions.NumericAssertionsExtensions.BeApproximately(NumericAssertions`1 parent, Single expectedValue, Single precision, String because, Object[] becauseArgs)
   at RoutingTests.UnitResponseModuleTests.U010_CheckTranslationAsync(Int64 sId, String sheetName, Int32 expected, Int32 excelRowCount) in C:\Users\sogscx\source\repos\Float\Compute\Test\RoutingTests\UnitResponseModuleTests.cs:line 480
   at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.BlockUntilCompleted()
```

### Versions

* Which version of Fluent Assertions are you using?
latest
* Which .NET runtime and version are you targeting? E.g. .NET framework 4.6.1 or .NET Core 2.1.
.NET Core 3.1 running NUnit

**Comment by jnyrup** [comment]
Please follow the Issue template
> Considered if this is a general question and not a bug?. For general questions please use StackOverflow.

Feel free to re-open if this turns out to be a bug and not a how to use the library.

**Comment by Smurf-IV** [comment]
Sorry missed that line.

**Comment by Smurf-IV** [comment]
Tracking:
https://stackoverflow.com/questions/63882877/how-do-you-use-beapproximately-with-a-value-that-changes-in-each-iteration-of

**Comment by Smurf-IV** [comment]
I cannot re-open, but as per SO. 
There needs to be argument validation for the precision so that -ves (Or zero) are not used for the tolerance value.

**Comment by jnyrup** [comment]
@Smurf-IV The fix will be available in the upcoming 6.0 release.

I've added guard against negative values, but not zeroes.
Approximating a value with a zero precision would of course be more clearly written as an exact comparison, but it works as expected.
